cmds/tui.c

```diff
@@ -2346,12 +2362,149 @@ static void tui_window_help(void)
 	delwin(win);
 }
 
+static void display_graph_field(WINDOW *win, int *selected_field, bool *graph_field_flag)
+{
+	int i;
+
+	for (i = 0; i < NUM_GRAPH_FIELD; i++) {
+		if (i == *selected_field)
+			wattron(win, A_REVERSE);
+		mvwprintw(win, i + ARRAY_SIZE(graph_field_help) + 4, 2, "[ %c ] %s", graph_field_flag[i] ? 'x' : ' ', fields[i]);
+		wattroff(win, A_REVERSE);
+	}
+}
+
+static void update_graph_output_fields(bool *graph_field_flag) {
+	struct display_field *field;
+	int i;
+	bool exist;
+
+	for (i = 0; i < NUM_GRAPH_FIELD; i++) {
+		exist = false;
+		list_for_each_entry(field, &graph_output_fields, list) {
+			if (field->id == i && !graph_field_flag[i]) {
+				exist = true;
+				del_field(field);
+				break;
+			}
+		}
+
+		if (!exist && graph_field_flag[i])
+			add_field(&graph_output_fields, graph_field_table[i]);
+	}
```

Comment:
This code can make the output fields displayed in a different order.  Say, 1. select total and self at first (press ENTER), 2. uncheck total (press ENTER), and finally add total back (press ENTER).  Now total time field comes after the self time.

To preserve the field order, how about this? (not tested)

```c
struct display_field *field, *tmp;
int i;

list_for_each_entry_safe(field, tmp, &graph_output_fields, list)
    del_field(field);

for (i = 0; i < NUM_GRAPH_FIELD; i++) {
    if (graph_field_flags[i])
        add_field(...);
}
```